docs: clarify the different way to build cln (#5479)
ElementsProject/lightning: doc/INSTALL.md

	sudo apt-get install -y cargo rustfmt

> There are two ways to build core lightning, and this depends on how you want use it.

Reviewer: ```suggestion
There are two ways to build core lightning, depending on how you want use it.
```

> To build cln to just install a tagged or master version you can use the following commands:

Reviewer: ```suggestion
To just install a tagged or master version you can build CLN with the following commands:
```

    pip3 install --upgrade pip
    pip3 install mako mistune==0.8.4 mrkd

>     ./configure RUST=1

Reviewer: ```suggestion
    ./configure --enable-rust
```

>     ./configure RUST=1

Author: Yep, I was thinking to make this explicit but I think this is not necessarily a make the docs confusing!

    make
    sudo make install

> N.B: if you want to disable Rust (a dependency of the grpc-plugin), you can set `RUST=0`

Reviewer: ```suggestion
N.B: if you want to disable Rust (a dependency of the grpc-plugin), you use `./configure --disable-rust`
```

> N.B: if you want disable the rust because you do not want use it or simple you do not want the grpc-plugin, you can set `RUST=0`

Reviewer: ```suggestion
N.B: If you want to disable Rust (a dependency of the grpc-plugin), you can set `RUST=0`
```

or just avoid to put `RUST` in front of configure and guess what the value will be (depending if you have rust installed or not).

To build core lightning for development purpose you can use the following commands:

    pip3 install poetry
    poetry shell
    poetry install
    ./configure --enable-developer
    make

>     make check DEVELOPER=1 EXPERIMENTAL_DUAL_FUND=0 VALGRIND=0

Author: `EXPERIMENTAL_DUAL_FUND` is a mistake, I will remove also `DEVELOPER`

> optionaly you can consider to use the `-j$(nproc)` in front of the `make` command to speed up the compilation.

Reviewer: ```suggestion
Optionally, you can use `make -j$(nproc)` to speed up the compilation.
```
